wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/schema/DevfileSchemaProvider.java

```diff
@@ -11,35 +11,48 @@
  */
 package org.eclipse.che.api.workspace.server.devfile.schema;
 
-import static org.eclipse.che.api.workspace.server.devfile.Constants.SCHEMA_LOCATION;
+import static org.eclipse.che.api.workspace.server.devfile.Constants.SCHEMAS_LOCATION;
+import static org.eclipse.che.api.workspace.server.devfile.Constants.SCHEMA_FILENAME;
 import static org.eclipse.che.commons.lang.IoUtil.getResource;
 import static org.eclipse.che.commons.lang.IoUtil.readAndCloseQuietly;
 
 import java.io.IOException;
 import java.io.StringReader;
 import java.lang.ref.SoftReference;
+import java.util.HashMap;
+import java.util.Map;
 import javax.inject.Singleton;
 
 /** Loads a schema content and stores it in soft reference. */
 @Singleton
 public class DevfileSchemaProvider {
 
-  private SoftReference<String> schemaRef = new SoftReference<>(null);
+  private Map<String, SoftReference<String>> schemas = new HashMap<>();
 
-  public String getSchemaContent() throws IOException {
-    String schema = schemaRef.get();
-    if (schema == null) {
-      schema = loadFile();
-      schemaRef = new SoftReference<>(schema);
+  public String getSchemaContent(String version) throws IOException {
+    if (schemas.containsKey(version)) {
+      String schema = schemas.get(version).get();
+      if (schema != null) {
+        return schema;
+      } else {
+        return loadAndPut(version);
+      }
+    } else {
+      return loadAndPut(version);
     }
-    return schema;
   }
 
-  public StringReader getAsReader() throws IOException {
-    return new StringReader(getSchemaContent());
+  public StringReader getAsReader(String version) throws IOException {
+    return new StringReader(getSchemaContent(version));
+  }
+
+  private String loadFile(String version) throws IOException {
+    return readAndCloseQuietly(getResource(SCHEMAS_LOCATION + version + "/" + SCHEMA_FILENAME));
```

Comment:
if bad version is passed, there will be a ugly message like "File not found...etc" maybe it is make sense to prettify it somehow, wdyt?